karibu-dsl/src/main/kotlin/com/github/mvysny/karibudsl/v10/Router.kt

```diff
@@ -17,6 +18,17 @@ public fun (@VaadinDsl HasComponents).routerLink(icon: VaadinIcon? = null, text:
     return link
 }
 
+@JvmName("routerLinkWithParams")
+@VaadinDsl
+public fun (@VaadinDsl HasComponents).routerLink(icon: VaadinIcon? = null, text: String? = null, viewType: KClass<out Component>,
+                                                        parameters: Map<String, String>, block: (@VaadinDsl RouterLink).() -> Unit = {}): RouterLink {
```

Comment:
In theory this could clash with the other routerLink() function (the routerLinkWIthParam one) if T would be a Map... but that kind of use-case would probably be extremely rare since the param types is 99% either String or Number.

Let's go ahead and merge the PR.